include/mata/nfa/delta.hh

```diff
@@ -97,6 +110,10 @@ public:
 class Delta {
 private:
     std::vector<Post> posts;
+    //this internal vector of things is done in a different way in each of the structures, post, delta, move.
+    //would be nice to make it uniform
+    //the delta way is the cleanest, no? a private data member
+    //lets do it that way?
 
```

Comment:
I am guessing that some of these classes will be extended in some version of mata, for instance to add some registers to automata.

Even Ii just inheritance is ok, it is weird to use three different ways for doing it for the three levels of delta, feels like a chaos, no? 
Then, if we want to choose one, then we should probably choose the most standard and robust one, which is the member data I think. ? The added boiler plate code wold be small.

Reply:
The added boilerplate code is extensive: about 20 functions for each class with a class member, only calling the underlying member variable methods. But I am OK with the boilerplate in case we choose this route. I agree that it is probably the most standard approach.

Anyone else has any opinions?

Reply:
In Post, each line with "using super::f" is replaced by the call of the method f of the member, the same number of lines, but longer.

Reply:
To clarify, the using-declaration is more concise, but more importantly, it includes all overloads of the function. Therefore, for each overload of a function in the base class, you need only one using-declaration in the derived class. It is a subtle different, but it comes in handy for class constructors, assignment operators, etc.

Reply:
So, two question:
Do we want that to be consistent across delta, post, and move (or their new names)?
If not, then fuck it, but I would be more for having it consistent, otherwise I will keep thinking about it whenever I see it.
If yes, which option? I guess members, since it feels robust and uncomplicated ?

Reply:
I would go with consistency and members when we are talking about Mata. It is the simplest approach, most robust, easier to modify, even if more verbose, cluttering the code files. But we have a separate file for delta, and we can split it even further if it comes to that.

Reply:
I agree, let's be consistent and use members.